taskcat/lambda_build.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+# authors:
+# Tony Vattathil <tonynv@amazon.com>, <avattathil@gmail.com>
+# Santiago Cardenas <sancard@amazon.com>, <santiago[dot]cardenas[at]outlook[dot]com>
+# Shivansh Singh <sshvans@amazon.com>,
+# Jay McConnell <jmmccon@amazon.com>,
+# Andrew Glenn <andglenn@amazon.com>
+from __future__ import print_function
+
+
+import os
+from shutil import make_archive
+from taskcat.colored_console import PrintMsg
+
+
+class LambdaBuild(object):
+    """Zips contents of lambda source files.
+
+    """
+
+    zip_file_name = "lambda"
+
+    def __init__(self, source_path, output_path="../packages"):
+
+        cur_dir = os.path.abspath(os.curdir)
+        try:
+            self.source_path = os.path.abspath(source_path)
+            os.chdir(self.source_path)
+            self.output_path = os.path.abspath(output_path)
```

Comment:
I'd flip the ordering here.

From a code reading point of view, it helps to think of method logic as telling a story:
1. get some information or do some setup
1. execute the work
1. return some results or do some cleanup

w/ that in mind setting `self.output_path` doesn't depend in any way on the `chdir` statement above it. So we can set our ivars together and then move onto the work of the method.

I also have some questions about whether we really need to set state (in the form of instance variables) for something that is so short lived and will not exist beyond the one call to the class. But that's more something to consider in the future tho not necessarily change now.

Reply:
@latrokles the `os.path.abspath` call is in fact dependent on the current directory (only if the supplied `output_path` is a relative path:

```
In [1]: import os

In [2]: os.curdir
Out[2]: '.'

In [3]: os.path.abspath(os.curdir)
Out[3]: '/private/tmp'

In [4]: os.chdir("test")

In [5]: os.path.abspath(os.curdir)
Out[5]: '/private/tmp/test'
```

Reply:
ohhhhh... I stand corrected. TIL